File: go-sdk/pkg/models/operations/getuserunpublishedarticles.go
package operations

import (
	"github.com/speakeasy-sdks/forem-sdks/go-client-sdk/pkg/models/shared"
)

type GetUserUnpublishedArticlesQueryParams struct {
	Page    *int32 `queryParam:"style=form,explode=true,name=page"`
	PerPage *int32 `queryParam:"style=form,explode=true,name=per_page"`
}

type GetUserUnpublishedArticlesRequest struct {
	QueryParams GetUserUnpublishedArticlesQueryParams
}

type GetUserUnpublishedArticlesResponse struct {
	ArticleIndices []shared.ArticleIndex
	ContentType    string
	StatusCode     int64
}

File: ts-sdk/src/sdk/models/operations/getuserunpublishedarticles.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetUserUnpublishedArticlesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;
}


export class GetUserUnpublishedArticlesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetUserUnpublishedArticlesQueryParams;
}


export class GetUserUnpublishedArticlesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: shared.ArticleIndex })
  articleIndices?: shared.ArticleIndex[];

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
